syndeo-template/src/test/java/com/google/cloud/syndeo/perf/SyndeoLoadTestUtils.java

```diff
@@ -0,0 +1,230 @@
+                .withInterval(Duration.millis(periodPerSplitMsecs)))
+        .apply(Reshuffle.viaRandomKey())
+        .apply(
+            FlatMapElements.into(TypeDescriptors.rows())
+                .via(
+                    inst -> {
+                      long ordinal =
+                          (inst.minus(Duration.millis(startTime.getMillis())).getMillis()
+                              / periodPerSplitMsecs);
+                      return LongStream.range(
+                              ordinal * MAX_ROWS_PER_SPLIT, (ordinal + 1) * MAX_ROWS_PER_SPLIT)
+                          .mapToObj(
+                              intVal ->
+                                  SyndeoLoadTestUtils.randomRowForSchema(
+                                      SyndeoLoadTestUtils.SIMPLE_TABLE_SCHEMA, 0.05, randomSeed))
+                          .collect(Collectors.toList());
+                    }))
+        .setRowSchema(SIMPLE_TABLE_SCHEMA);
+  }
+
+  // A schema for a table that has been slightly inspired on the Github public dataset
+  // provided by BigQuery.
+  public static final Schema SIMPLE_TABLE_SCHEMA =
+      Schema.builder()
+          .addField(Schema.Field.nullable("commit", Schema.FieldType.STRING))
+          .addField(Schema.Field.nullable("repo_name", Schema.FieldType.STRING))
+          .addField(Schema.Field.of("parent", Schema.FieldType.array(Schema.FieldType.STRING)))
+          .addField(Schema.Field.of("commitDate", Schema.FieldType.DATETIME))
+          .addField(Schema.Field.nullable("message", Schema.FieldType.STRING))
+          .addInt64Field("linesAdded")
+          .addInt32Field("linesRemoved")
+          .addBooleanField("merged")
+          .addByteArrayField("sha1")
+          // A decimal field that means nothing but that we include for good measure : )
+          .addDecimalField("decimalForGoodMeasure")
+          .build();
+
+  public static final Schema NESTED_TABLE_SCHEMA =
+      Schema.builder()
+          .addFields(SIMPLE_TABLE_SCHEMA.getFields())
+          .addField(
+              Schema.Field.nullable(
+                  "author",
+                  Schema.FieldType.row(
+                      Schema.builder()
+                          .addField(Schema.Field.of("name", Schema.FieldType.STRING))
+                          .addField(Schema.Field.of("email", Schema.FieldType.STRING))
+                          .build())))
+          .build();
+
+  public static String randomString(Integer length, Random randomSeed) {
+    return randomSeed
+        .ints(length, 48, 122)
+        .collect(StringBuilder::new, StringBuilder::appendCodePoint, StringBuilder::append)
+        .toString();
+  }
+
+  private static <T> T generateOrNull(
+      SerializableFunction<Void, T> generator, double nullProbability, Random randomSeed) {
+    if (randomSeed.nextDouble() > nullProbability) {
+      return generator.apply(null);
+    } else {
+      return null;
+    }
+  }
+
+  public static Row randomRowForSchema(
```

Comment:
You could use the StreamingDataGenerator here. It allows you to generate data according to a user specified schema to a sink at a specified QPS. You can add the pubsublite sink to the template and use that here. 

That will reduce a lot of the code here. I have used that in all the performance tests for templates.